farmdata2/farmdata2_modules/resources/BannerComponent.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * A Vue component for a message banner that is displayed when a specific event (success/error/etc.) occurs.
+ * 
+ * <h3 class="subsection-title">data-cy attributes</h3>
+ * <table>
+ * <thead><tr><th>Value</th>        <th>Description</th></tr></thead>
+ * <tbody>
+ * <tr><td>banner-handler</td>   <td>div containing bootstrap alert banner</td></tr>
+ * <tr><td>banner-close</td>   <td>'X' button that closes the alert banner</td></tr>
+ * <tr><td>banner-message</td>   <td>The message displayed inside the alert banner</td></tr>
+ * </tbody>
+ * </table>
+ * 
+ * <div>
+ *  @vue-prop {Object} [message={"msg": "Hello, I am a banner alert.", "class": "alert alert-warning"}] - Contains the message and class for the alert banner
+ *  @vue-prop {Boolean} [visible=null] - Set to true to show the alert banner. False makes the banner invisible.
+ *  @vue-prop {Boolean} [timeout=null] - Sets whether or not the banner dismisses itself which disables the 'x' to close. Set to false by default.
+ * </div>
+ * 
+ * <div>
+ *  @vue-event visibility-update - Updates the visibility in the parent page when the child component has internal visibility updates. This means each page needs to implement a function that catches this emit and matches their visibility with the child's internal visibility. 
+ * </div>
+ * @module
+ */ 
+let BannerComponent = {
+    template: 
+        `<div data-cy='banner-handler' :class="bannerClass" 
+            ref="banner" role="alert" 
+            v-show="isVisible"
+            :style="{ position: ['-webkit-sticky', 'sticky'], top: [topNav]}">
+            <button data-cy='banner-close' type="button" class="close" 
+            aria-label="Close" 
+            @click="hideBanner" 
+            v-show="timeout==false">
+            <span aria-hidden="true">&times;</span>

```

Comment:
I'm a little unclear on the purpose of this line.

Reply:
I believe there was some CSS reason for the inclusion of this element. I'll look into it and see if it can be removed.

Reply:
This has to be left in there because it actually displays the 'x' that needs to be clicked on.